### PR TITLE
timebase correction

### PR DESCRIPTION
**Pull request by despair86**
- finally making bencode ILP32/LLP64 clean (long is 32 bits even in 64-bit windows!)
- need to arrange more testing on the avx2 stuff (it looks promising)
- set up CI for windows builds
- clang-format

### CODE REVIEW

File: daemon/main.cpp [hunk 1]
```diff
@@ -127,5 +155,8 @@ main(int argc, char *argv[])
     llarp_main_free(ctx);
   }
   exit(code);
+#ifdef _WIN32
+  ::WSACleanup();
```
Comment (majestrate) [inline daemon/main.cpp]: called after `exit()` ?

File: llarp/ev_win32.hpp [hunk 2]
```diff
@@ -85,6 +85,81 @@ namespace llarp
...
+      {
+        t->before_write(t);
+      }
+      ev_io::flush_write();
+    }
+
+    int
+    read(void* buf, size_t sz)
+    {
+      ssize_t ret = tuntap_read(tunif, buf, sz);
```
Comment (majestrate) [inline llarp/ev_win32.hpp]: qusetion, does `tuntap_read` include a packet info header?

File: daemon/main.cpp (hunk 1 above)
Comment (despair86) [reply daemon/main.cpp]: fixed

**Comment by majestrate** [comment]
merged :^)

**Pull request merged**